src/io/flutter/run/daemon/FlutterDaemonController.java

```diff
@@ -246,30 +245,20 @@ private static GeneralCommandLine createBazelRunner(@NotNull Project project,
     commandLine.setCharset(CharsetToolkit.UTF8_CHARSET);
     commandLine.setExePath(FileUtil.toSystemDependentName(launchingScript));
 
-    // TODO(devoncarew): Fix the launch script to accept more flags.
-    //commandLine.addParameters("--machine");
-
-    // TODO(devoncarew): Fix the launch script to accept more flags.
-    //if (deviceId != null) {
-    //  commandLine.addParameter("--device-id=" + deviceId);
-    //}
-
-    // TODO(devoncarew): Fix the launch script to accept more flags.
-    //if (startPaused) {
-    //  commandLine.addParameter("--start-paused");
-    //}
-
     // Set the mode.
     if (mode != RunMode.DEBUG) {
       commandLine.addParameters("--define", "flutter_build_mode=" + mode.name());
     }
 
-    // Additional arguments.
-    if (additionalArguments != null) {
-      final StringTokenizer argumentsTokenizer = new CommandLineTokenizer(additionalArguments);
-      while (argumentsTokenizer.hasMoreTokens()) {
-        commandLine.addParameter(argumentsTokenizer.nextToken());
+    // User specified additional arguments.
+    final CommandLineTokenizer argumentsTokenizer =
+      additionalArguments != null ? new CommandLineTokenizer(additionalArguments) : new CommandLineTokenizer("");
```

Comment:
Alternatively:

    final CommandLineTokenizer argumentsTokenizer = new CommandLineTokenizer(StringUtil.notNullize(additionalArguments));

Shorter but maybe not better! :man_shrugging: